### PR TITLE
change function to class and add some more features

### PR DESCRIPTION
**Pull request by KimlengHor**
Problem:

- Measurement: there is no drop list yet. users can just type in a string for the field. (will change in the future)
- The buttons are not clickable yet

### CODE REVIEW

File: client/src/Upload.js [hunk 1]
```diff
@@ -51,36 +54,55 @@ const styles = makeStyles(theme => ({
...
-      });
+            columns: [
+                { title: 'Name', field: 'name' },
+                { title: 'Amount', field: 'amount'},
+                { title: 'Measurement', field: 'measurement'}
+            ],
+            instructionColumns: [
+                { title: 'Steps', field: 'description' },
+            ],
+            data: [
```
Comment (dpham123) [inline client/src/Upload.js]: Can we rename this `ingredientData`?

File: client/src/Upload.js [hunk 2]
```diff
@@ -51,36 +54,55 @@ const styles = makeStyles(theme => ({
...
-        data: [
-          { name: 'Soy sauce', amount: 63, description: "First step" },
-          {
-            name: 'Sugar',
-            amount:90,
-            description: "Second step"
-          },
-        ],
-      });
+            columns: [
```
Comment (dpham123) [inline client/src/Upload.js]: Can we rename this `ingredientColumns`?

File: client/src/Upload.js [hunk 3]
```diff
@@ -51,36 +54,55 @@ const styles = makeStyles(theme => ({
...
-export default function Upload() {
+class Upload extends React.Component {
+    constructor(props) {
+        super(props);
+        
+        this.state = {
+
+            uid: "",
+            recipeName: "",
+            mealTypeArray: ["breakfast", "lunch", "dinner"],
```
Comment (dpham123) [inline client/src/Upload.js]: There are 6 meal types. See ZenHub for more details: https://app.zenhub.com/workspaces/delight-5d85617c3b0ef90001a402e4/issues/cs157a-team18/cs157a-team18/55.

Also, if you are unable to change this array in the format specified in the ZenHub, maybe you can make `mealTypeArray` store mealType objects?

```
mealTypeArray: [{id: 1, type: "breakfast"}, {id: 4, type: "lunch"}, {id: 3, type: "dinner"}...]
```

File: client/src/Upload.js (hunk 1 above)
Comment (KimlengHor) [reply client/src/Upload.js]: fixed

File: client/src/Upload.js (hunk 2 above)
Comment (KimlengHor) [reply client/src/Upload.js]: fixed

File: client/src/Upload.js (hunk 3 above)
Comment (KimlengHor) [reply client/src/Upload.js]: fixed

**Pull request merged**